server/src/main/java/org/opensearch/index/translog/listener/TranslogEventListener.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.translog.listener;
+
+/**
+ * The listener that gets fired on events related to {@link org.opensearch.index.translog.TranslogManager}
+ *
+ * @opensearch.internal
+ */
+public interface TranslogEventListener {
+
+    TranslogEventListener NOOP_TRANSLOG_EVENT_LISTENER = new TranslogEventListener() {
+    };
+
+    /**
+     * Invoked after translog sync operations
+     */
+    default void onTranslogSync() {}
+
+    /**
+     * Invoked after recovering operations from translog
+     */
+    default void onTranslogRecovery() {}
```

Comment:
`onEndTranslogRecovery()` would better show of the semantics (matching `onBeginTranslogRecovery()`)